### DISCUSSION TITLE
Generator for ApplicationNotification base class

### DISCUSSION THREAD

**Discussion by rafaelpivato** [excid3/noticed#341] [Ideas]
Create a generator **`noticed:application_notification`** to create a base class for notifications of a Rails application.

```ruby
class ApplicationNotification < Noticed::Base
  def database_format
    {
      'type': self.class.name,
      'params': params
    }
  end

  def action_format
    params
  end
end
```

It would be good if `noticed:notification` could be modified to inherit from this ApplicationNotification whenever it exists. On the other hand, a warning could also be emitted if you have notifications already created in order to update their super class.

**Comment by excid3** [comment]
Definitely like this idea. I had manually added an ApplicationNotification to JumpstartRails.com and it would be great ot have built-in.

**Comment by nbarthel** [comment]
Should probably split this out as an open issue in 2.2.2.   The generator works, but using something like this deletes the application_notifier:

`rails d noticed:notifier CommentNotifier`

The generator also continues to try and overwrite the existing file every time.
